core/tests/performance/smart_pool/src/smart_pool_benchmark.cpp

```diff
@@ -127,6 +127,36 @@ BENCHMARK_DEFINE_F(SmartPoolFixture, VectorMultiThreadOperate)(benchmark::State
     }
 }
 
+static void BM_SmartPoolCreateAndAcquireVector(benchmark::State &state)
+{
+    // Perform setup here
+    const int poolSize = 1;
+    const int vectorSize(state.range(0));
+
+    for (auto _ : state) {
+        // This code gets timed
+        kpsr::SmartObjectPool<std::vector<float>> smartPool("SmartPoolBenchmark",
+                                                            poolSize,
+                                                            [&](std::vector<float> &vec) {
+                                                                vec.resize(vectorSize);
+                                                            });
+        std::shared_ptr<std::vector<float>> event = std::move(smartPool.acquire());
+        benchmark::DoNotOptimize((*event)[0] += 1);
```

Comment:
The first element in the vector is now read and written in order to avoid compiler optimization of the object